build/config.props

```
@@ -32,17 +32,7 @@
     <VsTargetBranch Condition="'$(IsEscrowMode)' == 'true'">rel/d$(VsTargetMajorVersion).$(MinorNuGetVersion)</VsTargetBranch>
     <VsTargetChannel Condition="'$(IsEscrowMode)' == 'true'">int.d$(VsTargetMajorVersion).$(MinorNuGetVersion)</VsTargetChannel>
 
-    <!-- .NET Core SDK Insertion Logic -->
-    <!--We don't have .NET5 installed on CI, so we need to get version from CliVersionForBuilding and install it-->
-    <!--TODO: remove temporary workaround of using LockSDKVersion. Tracking issue: https://github.com/NuGet/Home/issues/9712 -->
-    <LockSDKVersion>true</LockSDKVersion>
-    <OverrideCliBranchForTesting Condition="'$(LockSDKVersion)' == 'false'"></OverrideCliBranchForTesting>
-    <OverrideCliBranchForTesting Condition="'$(LockSDKVersion)' == 'true'">master 5.0.100</OverrideCliBranchForTesting>
-    <CliVersionForBuilding>"master 5.0.100"</CliVersionForBuilding>
-    <CliBranchForTesting Condition="'$(OverrideCliBranchForTesting)' != ''">$(OverrideCliBranchForTesting)</CliBranchForTesting>
-    <CliBranchForTesting Condition="'$(OverrideCliBranchForTesting)' == ''">master</CliBranchForTesting>
-    <CliTargetBranches Condition="'$(OverrideCliTargetBranches)' != ''">$(OverrideCliTargetBranches)</CliTargetBranches>
-    <CliTargetBranches Condition="'$(OverrideCliTargetBranches)' == ''">master</CliTargetBranches>
+    <CliBranchForTesting Condition="'$(CliBranchForTesting)' == ''">release/5.0.2xx 5.0.200-servicing.21120.4</CliBranchForTesting>
```

How about adding a comment about the two formats of property CliBranchForTesting?
channel            -> get the latest version from the channel
channel version -> get the specified version from the channel

---

That's a great idea. I think the channel is irrelevant when the version is provided according to the docs. I'll add a few examples.

---

Added an explanation. lmk what you think.

---

How about adding something like: multiple versions are supported, if separate them with semicolon？

---

Technically that's supported (and I didn't remove it), but downloading multiple versions will lead to a weird behavior in the tests, only 1 SDK will be tested. 

I think that it might be misleading if I were to add it there?

---

```suggestion
    <CliBranchForTesting Condition="'$(CliBranchForTesting)' == ''">release/5.0.2xx:5.0.200-servicing.21120.4</CliBranchForTesting>
```
How about using colon to separate channel and channel version?

---

> Technically that's supported (and I didn't remove it), but downloading multiple versions will lead to a weird behavior in the tests, only 1 SDK will be tested.
> 
> I think that it might be misleading if I were to add it there?
I see. Getting two versions work in the test might need much more effort than just adding a version here.

---

> downloading multiple versions will lead to a weird behavior in the tests, only 1 SDK will be tested.

`dotnet.integration.tests` was changed so it will look for an SDK that matches the TFM the test is running as. Therefore, if `dotnet.integration.tests` multi-targets, we NEED to download/install multiple SDKs. Also, `dotnet.integration.tests` will try to match the SDK and runtime version. I can't remember what happens if the test's runtime version doesn't match any .NET SDK version found in the `cli/sdk/`.

Anyway, I just wanted to highlight this change that was made while we were preparing for .NET 5 last year. Previously Nikolche was correct, if there were multiple SDKs installed in `cli/sdk/` only one would be patched and things wouldn't work correctly. But our test project was modified so now has some different behaviour.

---

PR https://github.com/NuGet/NuGet.Client/pull/3387 has enabled:
1.Recognize multiple SDK versions under cli folder dynamically
2.Create temporary Global.json with each SDK version so that the tests could run for different SDK versions
Since the SDK version is dynamically recognized (not hardcoded), most work should have been done in this PR for testing with multiple SDK versions.

I'm not sure if there is anything else needs to be changed in future, but I guess that should be miner changes, eg. the current TFM in some tests of dotnet.integration.tests is not supported in the future. 

So I changed my mind, I agree with adding multiple versions in the comment.

---

I'm not sure I know what the takeaway is here. 

Do we want a comment or not?

---

I think adding a comment mention that we are capable of processing multiple versions of SDK here, or adding an example of multiple versions. Either one works.

---

Same here, added a comment.